Mathlib/NumberTheory/ModularForms/EisensteinSeries/UniformConvergence.lean

```diff
@@ -0,0 +1,360 @@
+    (hx : (x 0, x 1) ∈ box n) : (Complex.abs (((x 0 : ℂ) * z + (x 1 : ℂ)))) ^ (-k) ≤
+      (((r z) ^ (-k) * n ^ (-k))) := by
+  by_cases hn : n = 0
+  · rw [hn] at hx
+    simp only [box_zero, Finset.mem_singleton, Prod.mk_eq_zero] at hx
+    by_cases hk0 : k = 0
+    rw [hk0] at *
+    simp only [neg_zero, Real.rpow_zero, mul_one, le_refl]
+    rw [hx.1, hx.2]
+    simp only [Int.cast_zero, zero_mul, add_zero, map_zero]
+    have h1 : (0 : ℝ) ^ (-k) = 0 := by
+      rw [Real.rpow_eq_zero_iff_of_nonneg (by rfl)]
+      simp only [ne_eq, neg_eq_zero, hk0, not_false_eq_true, and_self]
+    simp only [h1, hn, CharP.cast_eq_zero, mul_zero, le_refl]
+  · have hx2 : x ≠ 0 := by
+      rw [mem_box_ne_zero_iff_ne_zero n x hx]
+      exact hn
+    simp only [Int.mem_box] at hx
+    rw [Real.rpow_neg (by apply apply_nonneg), Real.rpow_neg ((r_pos z).le),
+      Real.rpow_neg (Nat.cast_nonneg n), ← mul_inv, inv_le_inv]
+    simp only [← hx, Nat.cast_max]
+    convert (rpow_bound hk z x hx2)
+    · simp only [Nat.cast_max]
+    · apply Real.rpow_pos_of_pos
+      apply Complex.abs.pos (linear_ne_zero ![x 0, x 1] z ?_)
+      simp only [ne_eq, Matrix.cons_eq_zero_iff, Int.cast_eq_zero, Matrix.zero_empty, and_true,
+        not_and] at *
+      intro hg h1
+      have : x = ![x 0, x 1] := by
+        exact List.ofFn_inj.mp rfl
+      rw [this, hg, h1] at hx2
+      simp only [Matrix.cons_eq_zero_iff, Matrix.zero_empty, and_self, not_true_eq_false] at *
+    · apply mul_pos (Real.rpow_pos_of_pos (r_pos z) _)
+      apply Real.rpow_pos_of_pos
+      norm_cast
+      exact Nat.pos_of_ne_zero hn
+
+theorem eis_is_bounded_on_box (k n : ℕ) (z : ℍ) (x : Fin 2 → ℤ)
+    (hx : (x 0, x 1) ∈ box n) : (Complex.abs (((x 0 : ℂ) * z + (x 1 : ℂ)) ^ k))⁻¹ ≤
+      (Complex.abs ((r z) ^ k * n ^ k))⁻¹ := by
+  have := eis_is_bounded_on_box_rpow (Nat.cast_nonneg k) z n x hx
+  norm_cast at *
+  simp_rw [zpow_neg, ← mul_inv] at this
+  have H : |r z ^ k * ↑(n ^ k)| = r z ^ k * ↑(n ^ k) := by
+    refine abs_eq_self.mpr ?_
+    apply mul_nonneg (pow_nonneg (r_pos z).le _)
+    simp only [Nat.cast_pow, ge_iff_le, Nat.cast_nonneg, pow_nonneg]
+  simp only [map_pow, zpow_coe_nat, H]
+  simpa using this
+
+lemma r_lower_bound_on_slice {A B : ℝ} (h : 0 < B) (z : upperHalfPlaneSlice A B) :
+    r ⟨⟨A, B⟩, h⟩ ≤ r z.1 := by
+  have hz := z.2
+  simp only [slice_mem_iff, abs_ofReal, ge_iff_le] at hz
+  simp_rw [r]
+  apply min_le_min
+  · dsimp only
+    convert hz.2
+    have := abs_eq_self.mpr (UpperHalfPlane.im_pos z.1).le
+    convert this.symm
+  rw [Real.sqrt_le_sqrt_iff]
+  simp only [r1', div_pow, one_div]
+  rw [inv_le_inv, add_le_add_iff_right]
+  apply div_le_div (sq_nonneg _)
+  · simpa [even_two.pow_abs] using pow_le_pow_left (abs_nonneg _) hz.1 2
+  · positivity
+  · simpa [even_two.pow_abs] using pow_le_pow_left h.le hz.2 2
+  · positivity
+  · positivity
+  · apply (r1_pos z).le
+
+end bounding_functions
+
+section summability
```

Comment:
I think this entire section can be split out with little effort

Reply:
You mean into a separate PR?